common/src/chain/pow.rs

```diff
@@ -27,6 +28,8 @@ pub struct PoWChainConfig {
     allow_min_difficulty_blocks: bool,
     /// The lowest possible difficulty
     limit: Uint256,
+    /// The distance required to pass to allow spending the block reward
+    reward_maturity_distance: BlockDistance,
```

Comment:
Why is this specific to `PowChainConfig`? Would it be appropriate for the setting to be agnostic to the consensus engine used?

Reply:
Nope. It depends on the consensus algorithm. That's a point we can discuss some time later.

But the technicality of where to put it and where to pull it from is an open question. I used this because it's part of chain config.